### ISSUE TITLE
Editor: erratic undo behaviour

### ISSUE THREAD

**Issue by Copons** [Automattic/simplenote-electron#523]
#### Steps to reproduce
1. Open a note.
2. Write or paste some content.
3. Hit the browser/app "Undo" button.

#### What I expected
To "backtrack" the history of my insertions.

#### What happened instead
When manually writing new content, it's possible to undo only the last inserted "block", except the first letter. E.g. "abcdef" is undone up to the "a".
When pasting, the pasted content is not undoable.
When part of a content is automatically added (e.g. the "-" of a list item), that is not undoable.

#### OS version
Windows App / Windows 10
Chrome 56 / macOS 10.12

#### Screenshot / Video
![https://cldup.com/dxaVb6z6f0.gif](https://cldup.com/dxaVb6z6f0.gif)

#### Additional notes
- This only applies for the "Undo" button (both in the toolbar and in the mouse context menu) provided by the browser or by the application. Hitting `cmd/ctrl+z` effectively undoes the changes.
- This does not happen in the macOS app.

**Comment by msilbers** [comment]
I had a report of this in #1322822-zen regarding the items mentioned above, as well as undoing anything with Hebrew text. The workaround mentioned above (Ctrl/Cmd+Z) still does work.

**Comment by msilbers** [comment]
On the Mac Electron app, Undo does not seem to do anything. It still works on the native MacOS app.

To replicate:
- delete some text
- hit Cmd + Z or go to Edit > Undo

Nothing will happen! Then, I hit enter twice after a URL. Hitting Undo pasted the URL to a new line, very unexpected behavior. In the following gif, each time the menu comes down I am clicking **Undo** but I couldn't capture that part:

![simplenote-undo-errors](https://user-images.githubusercontent.com/31680407/50425918-97471480-084e-11e9-8c19-b4bff101c519.gif)

I was able to replicate this a few times

**Comment by rachelmcr** [comment]
I noticed today that the Undo button is very erratic, sometimes even mangling the note content. Here's an example where using Undo to go back to a previous line ends up adding content to the next line of the note:

![electron_undo-button](https://user-images.githubusercontent.com/8658164/79964294-65fe4900-8482-11ea-9fe0-e74927b32727.gif)

I noticed that the Redo button won't redo every change from the Undo button (it seems to stop after redoing all the changes on the current line).

I also noticed that the Undo/Redo buttons and the Undo/Redo keyboard shortcuts don't work well together. If I use the Undo button to undo typing, I can't use the Redo keyboard shortcut to revert that change — I actually have to use the Undo keyboard shortcut to revert it.
